beancount-stubs/core/data.pyi

```diff
@@ -129,8 +130,8 @@ class Transaction:
     flag: Flag
     payee: Optional[str]
     narration: str
-    tags: Optional[set]
-    links: Optional[set]
+    tags: FrozenSet
```

Comment:
only `EMPTY_SET` is a frozen set - probably to make sure it stays empty :) 

The Transaction's definition specifies `Set` [here](https://github.com/beancount/beancount/blob/v2/beancount/core/data.py#L224) and the "sanity checker" accepts both `Set` and `FrozenSet` [here](https://github.com/beancount/beancount/blob/be13ab359987947eb0f8d241fde55adc65cb4dbc/beancount/core/data.py#L498).

Reply:
Ah, joke's on me—when I said "matches upstream code" I was looking at [`master`](https://github.com/beancount/beancount/blob/664e4b32302d339c4104db8570448d3ee9a99d3e/beancount/core/data.py#L250-L251).

How about:
```suggestion
    tags: FrozenSet | set
```

Or, although [that `allow_none_for_tags_and_links` switch](https://github.com/beancount/beancount/blob/be13ab359987947eb0f8d241fde55adc65cb4dbc/beancount/core/data.py#L499C16-L499C45) defaults to `False`:
```suggestion
    tags: FrozenSet | set | None
```

Reply:
I'd say let's go with `None` being a part of the set. And no worries about the branch mismatch, been in the same position a week ago, writing my own importer... :)

Reply:
Actually, since I'm annotating args, [the docs](https://docs.python.org/3.8/library/typing.html#typing.Set) say I should use `AbstractSet`.
```python
>>> from typing import AbstractSet
>>> issubclass(set, AbstractSet)
True
>>> issubclass(frozenset, AbstractSet)
True
```
Although `AbstractSet` is [deprecated since 3.9](https://docs.python.org/3.12/library/typing.html#typing.AbstractSet), most of `typing` is too, so I won't worry about that yet.

Should I fix `Document` accordingly in this or another PR?